File: api/src/main/java/dev/vibris/api/ScenePreset.java
package dev.vibris.api;

import java.util.Objects;

public record ScenePreset(String presetId, String displayName, SceneContext context) {
    public ScenePreset {
        presetId = Objects.requireNonNull(presetId, "presetId");
        displayName = Objects.requireNonNull(displayName, "displayName");
        context = Objects.requireNonNull(context, "context");
        if (presetId.isBlank() || displayName.isBlank()) throw new IllegalArgumentException("Preset names are blank");
    }
}

File: api/src/main/kotlin/dev/vibris/api/ScenePreset.kt
package dev.vibris.api

@JvmRecord
data class ScenePreset(
    val presetId: String,
    val displayName: String,
    val context: SceneContext,
    val version: String,
) {
    constructor(presetId: String, displayName: String, context: SceneContext) :
        this(presetId, displayName, context, "1")

    init {
        require(presetId.isNotBlank() && displayName.isNotBlank() && version.isNotBlank()) {
            "Preset names are blank"
        }
    }
}